cg/meta/demultiplex/housekeeper_storage_functions.py

```diff
@@ -203,14 +208,42 @@ def add_file_to_bundle_if_non_existent(
             file=file_path,
             tags=tag_names,
         )
+        LOG.info(f"Sample sheet added to Housekeeper bundle {bundle_name}")
+    else:
+        LOG.info(f"Bundle {bundle_name} already has a sample sheet")
 
 
 def file_exists_in_latest_version_for_bundle(
     file_path: Path, bundle_name: str, hk_api: HousekeeperAPI
 ) -> bool:
-    """Check if file exists in latest version for bundle."""
+    """Check if file exists in the latest version for bundle."""
     latest_version: Version = hk_api.get_latest_bundle_version(bundle_name=bundle_name)
 
     return any(
         file_path.name == Path(bundle_file.path).name for bundle_file in latest_version.files
     )
+
+
+def get_sample_sheets_from_latest_version(flow_cell_id: str, hk_api: HousekeeperAPI) -> List[File]:
+    """Returns the files tagged with 'samplesheet' or 'archived_sample_sheet' for the given bundle."""
+    try:
+        files: List[File] = hk_api.get_files_from_latest_version(
+            bundle_name=flow_cell_id, tags=[flow_cell_id]
+        ).all()
+        sample_sheet_files: List[File] = filter_on_sample_sheets(files=files)
+    except HousekeeperBundleVersionMissingError:
+        sample_sheet_files: List = []
+    return sample_sheet_files
```

Comment:
I would prefer something like this, and remove the filtering function.
```suggestion
def get_sample_sheets_from_latest_version(flow_cell_id: str, hk_api: HousekeeperAPI) -> List[File]:
    """Returns the files tagged with 'samplesheet' or 'archived_sample_sheet' for the given bundle."""
    try:
        sheets_with_normal_tag: List[File] = hk_api.get_files_from_latest_version(
            bundle_name=flow_cell_id, tags=[flow_cell_id,SequencingFileTag.SAMPLE_SHEET]
        ).all()
        sheets_with_archive_tag List[File]  =hk_api.get_files_from_latest_version(
            bundle_name=flow_cell_id, tags=[flow_cell_id,SequencingFileTag. ARCHIVED_SAMPLE_SHEET]
        ).all()
        sample_sheet_files: List[File] = filter_on_sample_sheets(files=files)
    except HousekeeperBundleVersionMissingError:
        sample_sheet_files: List = []
    return sheets_with_normal_tag + sheets_with_archive_tag
```

Reply:
fixed